src/Gii.php

```diff
@@ -13,31 +12,24 @@
 final class Gii implements GiiInterface
 {
     /**
-     * @param array<string, GeneratorInterface|LazyGenerator> $generators
+     * @param array<string, GeneratorInterface|GeneratorProxy> $proxies
      */
-    public function __construct(private array $generators)
+    public function __construct(private array $proxies)
     {
     }
 
     public function addGenerator(GeneratorInterface $generator): void
     {
-        $this->generators[$generator::getId()] = $generator;
+        $this->proxies[$generator::getId()] = new GeneratorProxy(fn () => $generator, $generator::class);
     }
 
     public function getGenerator(string $id): GeneratorInterface
     {
-        if (!isset($this->generators[$id])) {
-            throw new GeneratorNotFoundException('Generator "' . $id . '" not found');
-        }
-
-        return $this->generators[$id] instanceof Closure ? $this->generators[$id]() : $this->generators[$id];
+        return $this->proxies[$id]?->loadGenerator() ?? throw new GeneratorNotFoundException('Generator "' . $id . '" not found');
```

Comment:
Proxies can contain ready-to-use generators (`array<string, GeneratorInterface|GeneratorProxy> $proxies`).

Reply:
Done